code/modules/client/preferences.dm

```diff
@@ -492,3 +500,226 @@ GLOBAL_LIST_EMPTY(preferences_datums)
+						slot_id,
+					)
+					last_import_error_map[PREFERENCE_IMPORT_ERROR_NOT_ENOUGH_CHARACTER_SLOTS] = new_value
+					continue
+
+			var/list/character_data_raw = raw_data[data_key]
+			load_character(slot_id)
+			for(var/character_data_key in character_data_raw)
+				if(handle_import_preference(character_data_key, character_data_raw[character_data_key], PREFERENCE_CHARACTER))
+					continue
+				last_import_error_map[PREFERENCE_IMPORT_ERROR_UNKNOWN_CHARACTER_DATA_KEY] += list(character_data_key)
+
+			save_character()
+			continue
+
+		if(handle_import_preference(data_key, raw_data[data_key], PREFERENCE_PLAYER))
+			continue
+
+		if(handle_import_raw(data_key, raw_data[data_key]))
+			continue
+
+		last_import_error_map[PREFERENCE_IMPORT_ERROR_UNKNOWN_PLAYER_DATA_KEY] += list(data_key)
+
+	// save and load to trigger full refresh on preferences
+	save_preferences()
+	load_preferences()
+
+/// Handles importing data that isnt managed using preferences
+/datum/preferences/proc/handle_import_raw(key, data)
+	// keys that should not be overwritten when imported
+	var/static/list/ignored_keys = list(
+		"muted",
+		"lastchangelog",
+		"hearted_until",
+		"version",
+	)
+
+	if(key in ignored_keys)
+		return TRUE
+
+	switch(key)
+		if("default_slot")
+			default_slot = sanitize_integer(data, 1, max_save_slots, initial(default_slot))
+			return TRUE
+
+		if("chat_toggles")
+			chat_toggles = sanitize_integer(data, NONE, ~NONE, initial(toggles))
+			return TRUE
+
+		if("toggles")
+			toggles = sanitize_integer(data, NONE, ~NONE, initial(toggles))
+			return TRUE
+
+		if("ignoring")
+			ignoring = sort_list(SANITIZE_LIST(data))
+			if(length(ignoring) >= 255)
+				ignoring.len = 255
+			return TRUE
+
+		if("be_special")
+			be_special = sanitize_be_special(SANITIZE_LIST(data))
+			return TRUE
+
+		if("favorite_outfits")
+			var/list/parsed_favs = list()
+			for(var/typetext in SANITIZE_LIST(data))
+				var/datum/outfit/path = text2path(typetext)
+				if(ispath(path)) //whatever typepath fails this check probably doesn't exist anymore
+					parsed_favs += path
+			favorite_outfits = unique_list(parsed_favs)
+			return TRUE
+
+		if("key_bindings")
+			key_bindings = sanitize_keybindings(SANITIZE_LIST(data))
+			key_bindings_by_key = get_key_bindings_by_key(key_bindings)
+			return TRUE
+
+/// Handles importing special preferences for characters
+/datum/preferences/proc/handle_specialized_character_preference_import(key, data)
+	switch(key)
+		if("version") // we already checked this
+			return TRUE
+
+		if("randomise")
+			randomise = list()
+			var/datum/preference_middleware/random/randomization_handler = locate() in middleware
+			data = SANITIZE_LIST(data)
+			for(var/randomization_key in data)
+				if(!randomization_handler.set_random_preference(randomization_key, data[randomization_key], parent.mob))
+					last_import_error_map[PREFERENCE_IMPORT_ERROR_INVALID_VALUE_RANDOMIZATION_KEY] += list(randomization_key)
+			return TRUE
+
+		if("all_quirks")
```

Comment:
Does this perform any validation on if the quirks are valid? It doesn't look like it does